clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -59,6 +59,7 @@
     private final SelectionKey key;
     private final SocketChannel socketChannel;
     private final Logger log;
+    private final ByteBuffer emptyBuf = ByteBuffer.allocate(0);
```

Comment:
Why not just use the static value `ByteUtils#EMPTY_BUF` here?  All empty buffers are the same, and they can never be enlarged or changed.